lib_xua/src/core/buffer/ep/ep_buffer.xc

```diff
@@ -444,114 +445,102 @@ void XUA_Buffer_Ep(
                 break;
             }
 #endif
-            /* Sample Freq or chan count update from Endpoint 0 core */
-            case testct_byref(c_aud_ctl, u_tmp):
+            /* Sample Freq or stream format update (e.g. channel count) from Endpoint 0 core */
+            case inct_byref(c_aud_ctl, cmd):
             {
-                if (u_tmp)
+                if(cmd == SET_SAMPLE_FREQ)
                 {
-                   // is a control token sent by reboot_device
-                   inct(c_aud_ctl);
-                   outct(c_aud_ctl, XS1_CT_END);
-                   while(1) {};
-                }
-                else
-                {
-                    unsigned cmd = inuint(c_aud_ctl);
-
-                    if(cmd == SET_SAMPLE_FREQ)
-                    {
-                        unsigned receivedSampleFreq = inuint(c_aud_ctl);
+                    unsigned receivedSampleFreq = inuint(c_aud_ctl);
```

Comment:
Just counting ins and outs in this diff and it looks like the old one was doing more outs than the new one. Just flagging to make sure this is the intended behavior and not a deadlock.